src/services/blocks/BlocksService.ts

```diff
@@ -264,40 +265,50 @@ export class BlocksService extends AbstractService {
 				previousBlockHash
 			);
 
-			const doesQueryFeeDetailsExist = this.hasQueryFeeApi.hasQueryFeeDetails(
-				specVersion.toNumber()
+			const transactionPaidFeeEvent = xtEvents.find(
+				({ method }) =>
+					isFrameMethod(method) && method.method === Event.transactionPaidFee
 			);
+
 			let finalPartialFee = partialFee.toString(),
 				dispatchFeeType = 'preDispatch';
-			/**
-			 * Call queryFeeDetails. It may not be available in the runtime and will
-			 * error automatically when we try to call it. We cache the runtimes it will error so we
-			 * don't try to call it again given a specVersion.
-			 */
-			if (doesQueryFeeDetailsExist === 'available') {
-				finalPartialFee = await this.fetchQueryFeeDetails(
-					block.extrinsics[idx].toHex(),
-					previousBlockHash,
-					weightInfo.weight,
-					weight
-				);
-
+			if (transactionPaidFeeEvent) {
+				finalPartialFee = transactionPaidFeeEvent.data[1].toString();
 				dispatchFeeType = 'postDispatch';
```

Comment:
Wondering whether adding a new `dispatchFeeType` for this (eg `fromEvent`) makes sense?

Why? because the queryFee logic we use is hardcoded to follow the polkadot fee payment approach and may not work in chains that do something different, so adding a new dispatchFeeType allows chains to know whether they can rely on the value or not if they differ.

Reply:
Ahh that makes complete sense. I think this is valid, and I agree that a `fromEvent` `dispatchFeeType` is a nice addition.

Reply:
Added `fromEvent` dispatchFeeType